academy/xl-cosmjs/multi-msg.md

```diff
@@ -0,0 +1,508 @@
+
+</HighlightBox>
+
+## Multiple token transfer messages
+
+From here, you can add an extra message for a token transfer from Alice to Bob:
+
+```typescript
+const result = await signingClient.signAndBroadcast(
+    alice,
+    [
+        {
+            typeUrl: "/cosmos.bank.v1beta1.MsgSend",
+            value: {
+                fromAddress: alice,
+                toAddress: faucet,
+                amount: [
+                    { denom: "uatom", amount: "100000" },
+                ],
+            },
+          },
+        {
+            typeUrl: "/cosmos.bank.v1beta1.MsgSend",
+            value: {
+                fromAddress: alice,
+                toAddress: bob,
+                amount: [
+                    { denom: "token", amount: "10" },
+                ],
+            },
+          },
+    ],
+    "auto",
+)
+```
+
+## Mixing other message types
+
+The above example shows you two token-transfer messages in a single transaction. You can see this with their `typeUrl: "/cosmos.bank.v1beta1.MsgSend"`.
+
+Neither Cosmos nor CosmJS limit you to messages of the same type. You can decide to have other message types along a token transfer. For instance, in one transaction Alice could:
+
+1. Send tokens to the faucet.
+2. Delegate some of her tokens to Bob the validator.
+
+How would Alice create the second message? Find your message type in `SigningStargateClient`'s registry of types:
+
+```typescript [https://github.com/cosmos/cosmjs/blob/7aad551/packages/stargate/src/signingstargateclient.ts#L94]
+    ["/cosmos.staking.v1beta1.MsgDelegate", MsgDelegate],
+```
+
+Click through to the type definition, and the `cosmjs-types` repository:
+
+```typescript [https://github.com/confio/cosmjs-types/blob/a14662d/src/cosmos/staking/v1beta1/tx.ts#L46-L50]
+export interface MsgDelegate {
+    delegatorAddress: string;
+    validatorAddress: string;
+    amount?: Coin;
+}
+```
+
+With this information, Alice sends the following transaction:
+
+```typescript
+const result = await signingClient.signAndBroadcast(
+    alice,
+    [
+        {
+            typeUrl: "/cosmos.bank.v1beta1.MsgSend",
+            value: {
+                fromAddress: alice,
+                toAddress: faucet,
+                amount: [
+                    { denom: "uatom", amount: "100000" },
+                ],
+            },
+        },
+        {
+            typeUrl: "/cosmos.staking.v1beta1.MsgDelegate",
+            value: {
+                delegatorAddress: alice,
+                validatorAddress: bob,
+                amount: { denom: "uatom", amount: "1000", },
+            },
+          },
+    ],
+    "auto",
+)
+```
+
+It is also possible to put more than two messages in a single transaction, as long as your raw transaction has fewer bytes than the limit.
+
+When you create [your own message types in CosmJS](TODO), they have to follow this format and be declared in the same fashion.
+
```

Comment:
We have the Keplr tutorial already, so might be nice to add a call to action here :)